pkg/trace/writer/trace.go

```diff
@@ -20,6 +20,13 @@ import (
 
 const pathTraces = "/api/v0.2/traces"
 
+// payloadMaxSize is the maximum payload size accepted by the Datadog API, after unpacking
+const payloadMaxSize = int(3.2 * 1024 * 1024) // 3.2MB
+
+// payloadFlushThreshold specifies the maximum accumulated payload size that is allowed before
+// a flush is triggered; replaced in tests.
+var payloadFlushThreshold = int(payloadMaxSize * 60 / 100) // 60% of max
```

Comment:
> Do we really need both payloadMaxSize and payloadFlushThreshold? Seems we could be fine with having just one MaxBytesPerPayload (analogous to the previous MaxSpansPerPayload)?

I think we should stay like this. It makes it clearly visible to the reader what the limit of the Datadog API is. Having just a "random" value will make things confusing. We do the same in the Go client.

> Also, should it be configurable like the previous max spans limit? Benefits:

Good points! I don't see why someone would want smaller payloads to be sent. Raising it however is a good point, but I doubt we'll change that any time soon. We could say the same about all our integrations where the flush threshold size is hard-coded. Was never a problem.

Let's see what others say, if people feel this should be configurable, we should do it.

> If not set in the config then it can default to this "slightly less than the current max at our API", and that value can be hardcoded here.

I chose to go around half the size, to reflect the choice that was made in our tracers as well, and mostly to allow some buffer for potential big package surprises.

Reply:
@LotharSee @vlad-mh @palazzem do you have opinions here? Adding the limit seems safe but normally should not be necessary. We've been doing this in our tracers for ages and it was never a problem.

Reply:
I don't think it's worth exposing this as configurable. additional work (+ required documentation) for something the user is unlikely to change. He can't increase it (as we have a limit in the backend), and decreasing it probably doesn't make a difference for him.

My 2 cents.

Reply:
Yeah makes sense. I was comparing to the max spans limit which was clearly useful to have configurable because it was so low, but if this is working as intended and getting the agent to send "large enough" payloads, then there's probably no reason to make it configurable.